Mathlib/RingTheory/AdicCompletion/Exactness.lean

```diff
@@ -0,0 +1,205 @@
+/-
+Copyright (c) 2024 Judith Ludwig, Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judith Ludwig, Christian Merten
+-/
+import Mathlib.Algebra.Exact
+import Mathlib.RingTheory.AdicCompletion.Functoriality
+import Mathlib.RingTheory.Filtration
+
+/-!
+# Exactness of adic completion
+
+In this file we establish exactness properties of adic completions. In particular we show:
+
+## Main results
+
+- `AdicCompletion.map_surjective`: Adic completion preserves surjectivity.
+- `AdicCompletion.map_injective`: Adic completion preserves injectivity
+  of maps between finite modules over a Noetherian ring.
+- `AdicCompletion.map_exact`: Over a noetherian ring adic completion is exact on finite
+  modules.
+
+## Implementation details
+
+All results are proven directly without using Mittag-Leffler systems.
+
+-/
+
+universe u v w t
+
+open LinearMap
+
+namespace AdicCompletion
+
+variable {R : Type u} [CommRing R] {I : Ideal R}
+
+section Surjectivity
+
+variable {M : Type v} [AddCommGroup M] [Module R M]
+variable {N : Type w} [AddCommGroup N] [Module R N]
+
```

Comment:
```suggestion
```

Reply:
This reduces universe generality for `map_surjective`, so I'd rather keep it the way it is, if that's okay. Note that the injectivity part currently needs stricter universe assumptions because of the Artin-Rees lemma only being proved in this case.

Reply:
Oh sorry, I didn't notice at all that the universes change. Not problem at all!